libcontainer/nsenter/nsexec.c

```diff
@@ -854,22 +886,29 @@ void nsexec(void)
 					bail("failed to sync with parent: SYNC_USERMAP_ACK: got %u", s);
 				/* Switching is only necessary if we joined namespaces. */
 				if (config.namespaces) {
-					if (prctl(PR_SET_DUMPABLE, 0, 0, 0, 0) < 0)
+					if (prctl(PR_SET_DUMPABLE, 0, 0, 0, 0) <
+					    0)
 						bail("failed to set process as dumpable");
 				}
 
 				/* Become root in the namespace proper. */
 				if (setresuid(0, 0, 0) < 0)
 					bail("failed to become root in user namespace");
 			}
-
-			/*
-			 * Unshare all of the namespaces. Note that we don't merge this
-			 * with clone() because there were some old kernel versions where
-			 * clone(CLONE_PARENT | CLONE_NEWPID) was broken, so we'll just do
-			 * it the long way.
+/*
+			 * Unshare all of the namespaces. Now, it should be noted that this
+			 * ordering might break in the future (especially with rootless
+			 * containers). But for now, it's not possible to split this into
+			 * CLONE_NEWUSER + [the rest] because of some RHEL SELinux issues.
+			 *
+			 * Note that we don't merge this with clone() because there were
+			 * some old kernel versions where clone(CLONE_PARENT | CLONE_NEWPID)
+			 * was broken, so we'll just do it the long way anyway.
 			 */
-			if (unshare(config.cloneflags) < 0)
+			uint32_t flags = config.cloneflags;
+			if (config.cloneflags & CLONE_NEWCGROUP)
+				flags &= ~CLONE_NEWCGROUP;
+			if (unshare(flags) < 0)
```

Comment:
I would just do `config.cloneflags & ~CLONE_NEWCGROUP` and not have the temporary `flags` variable. But that's just me...